### PR TITLE
rsyslog formula improvements

### PR DESCRIPTION
**Pull request by aboe76**
- Added defaults.yaml to include package defaults
 - Added distrospecific overwrites to map.jinja

 - Added imjournal mode

 - improved template and moved it to own directory to better seperate
   file and templates

### CODE REVIEW

**Comment by aboe76** [comment]
can somebody please take a look at merge this if they agree with these changes?

File: pillar.example [hunk 1]
```diff
@@ -11,8 +11,11 @@ rsyslog:
   listentcpprt: 10514                     # specify tcp listen port ($InputTCPServerRun)
   listenudp: true                         # omit to disable listening on udp port
   imkllog: true                           # omit to log kernel messages
+  imjourrnal: false                       # omit to log journal messages
```
Comment (babilen) [inline pillar.example]: This should be `imjournal` I believe.

**Comment by aboe76** [comment]
@babilen thanks for checking and finding the issue.

**Comment by babilen** [comment]
Looks better! Ta :)

**Comment by aboe76** [comment]
@babilen thanks

**Pull request merged**